pymc3/glm/linear.py

```diff
@@ -150,6 +158,27 @@ def __init__(self, x, y, intercept=True, labels=None,
             name='', y_est=self.y_est,
             y_data=y, model=self)
 
+    @staticmethod
+    def check_observed_vars_input(y):
+        if isinstance(y, (DataFrame, Series, np.ndarray)):
```

Comment:
`if not isinstance`...
And `pm.Dataframe`, `pm.Series`.
y can also be a theano.shared

Reply:
There is a `pass` here, so `if instance` is correct, but it's not relevant given your comment below.